test/bdd/pkg/edv/edv_steps.go

```diff
@@ -49,71 +73,325 @@ func (e *Steps) createDataVault(vaultID, expectedVaultLocation string) error {
+		err := getJWEFieldValues(retrievedJWEFields, "retrieved JWE")
+	if err != nil {
+		return err
+	}
+
+	err = verifyFieldsAreEqual(
+		retrievedProtectedFieldValue, expectedProtectedFieldValue,
+		retrievedIVFieldValue, expectedIVFieldValue,
+		retrievedCiphertextFieldValue, expectedCiphertextFieldValue,
+		retrievedTagFieldValue, expectedTagFieldValue)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getJWEFieldValues(jweFields map[string]string, jweDocType string) (string, string, string, string, error) {
+	protectedFieldValue, found := jweFields["protected"]
+	if !found {
+		return "", "", "", "", fieldNotFoundError("protected", jweDocType)
+	}
+
+	ivFieldValue, found := jweFields["iv"]
+	if !found {
+		return "", "", "", "", fieldNotFoundError("iv", jweDocType)
+	}
+
+	ciphertextFieldValue, found := jweFields["iv"]
+	if !found {
+		return "", "", "", "", fieldNotFoundError("ciphertext", jweDocType)
+	}
+
+	tagFieldValue, found := jweFields["tag"]
+	if !found {
+		return "", "", "", "", fieldNotFoundError("tag", jweDocType)
+	}
+
+	return protectedFieldValue, ivFieldValue, ciphertextFieldValue, tagFieldValue, nil
+}
+
+func verifyFieldsAreEqual(retrievedProtectedFieldValue, expectedProtectedFieldValue, retrievedIVFieldValue,
+	expectedIVFieldValue, retrievedCiphertextFieldValue, expectedCiphertextFieldValue, retrievedTagFieldValue,
+	expectedTagFieldValue string) error {
+	if retrievedProtectedFieldValue != expectedProtectedFieldValue {
+		return unexpectedValueError(expectedProtectedFieldValue, retrievedProtectedFieldValue)
+	}
+
+	if retrievedIVFieldValue != expectedIVFieldValue {
+		return unexpectedValueError(expectedIVFieldValue, retrievedIVFieldValue)
+	}
+
+	if retrievedCiphertextFieldValue != expectedCiphertextFieldValue {
+		return unexpectedValueError(expectedCiphertextFieldValue, retrievedCiphertextFieldValue)
+	}
+
+	if retrievedTagFieldValue != expectedTagFieldValue {
+		return unexpectedValueError(expectedTagFieldValue, retrievedTagFieldValue)
+	}
+
+	return nil
+}
+
+func verifyStructuredDocsAreEqual(decryptedDoc, expectedDoc operation.StructuredDocument) error {
+	if decryptedDoc.ID != expectedDoc.ID {
+		return unexpectedValueError(expectedDoc.ID, decryptedDoc.ID)
+	}
+
+	expectedCreatedValue, decryptedCreatedValue, err := getMetaFieldValues(expectedDoc, decryptedDoc)
+	if err != nil {
+		return err
+	}
+
+	expectedMessageValue, decryptedMessageValue, err := getContentFieldValues(expectedDoc, decryptedDoc)
+	if err != nil {
+		return err
+	}
+
+	if decryptedCreatedValue != expectedCreatedValue {
+		return unexpectedValueError(expectedCreatedValue, decryptedCreatedValue)
+	}
+
+	if decryptedMessageValue != expectedMessageValue {
+		return unexpectedValueError(expectedMessageValue, decryptedMessageValue)
 	}
 
 	return nil
 }
+
+func getContentFieldValues(expectedDoc, decryptedDoc operation.StructuredDocument) (string, string, error) {
+	expectedMessageFieldInContent, found := expectedDoc.Content["message"]
```

Comment:
use a const for `message` key in this function

Reply:
Done